### PR TITLE
B-1087 - Add accept project lead with rest

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: src/api/Project/queries.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { UseQueryProps, useBaseQuery } from "src/api/useBaseQuery";
+import { API_PATH } from "src/api/ApiPath";
+import { PROJECT_TAGS } from "./tags";
+import { components } from "src/__generated/api";
+
+export type useGetProjectBySlugResponse = components["schemas"]["ProjectResponse"];
```
Comment (haydencleary) [inline src/api/Project/queries.ts:6]:
```suggestion
export type UseGetProjectBySlugResponse = components["schemas"]["ProjectResponse"];
```

File: src/api/me/mutations.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { API_PATH } from "src/api/ApiPath";
+import { components } from "src/__generated/api";
+import { UseMutationProps, useBaseMutation } from "../useBaseMutation";
+import { PROJECT_TAGS } from "../Project/tags";
+
+export type useCreateProjectBody = components["schemas"]["CreateProjectRequest"];
+export type useCreateProjectResponse = components["schemas"]["CreateProjectResponse"];
```
Comment (haydencleary) [inline src/api/me/mutations.ts:7]:
```suggestion
export type UseCreateProjectBody = components["schemas"]["CreateProjectRequest"];
export type UseCreateProjectResponse = components["schemas"]["CreateProjectResponse"];
```

File: src/api/me/mutations.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+const useAcceptProjectLeaderInvitation = ({
+  options = {},
+  params,
+}: UseMutationProps<unknown, { projectId: string; projectSlug: string }, undefined>) => {
+  return useBaseMutation<undefined, unknown>({
+    resourcePath: API_PATH.ME_PROJECT_LEADER_INVITATIONS(params?.projectId || ""),
+    method: "PUT",
```
Comment (haydencleary) [inline src/api/me/mutations.ts:15]: Isn't it already `PUT` by default ?

File: src/components/ProjectLeadInvitation/ProjectLeadInvitation.tsx [hunk 4]
```diff
@@ -1,32 +1,28 @@
...
+}: ProjectLeadInvitationProps) {
+  const { mutate } = MeApi.mutations.useAcceptProjectLeaderInvitation({ params: { projectId, projectSlug } });
 
-  const { data } = useGetProjectLeadInvitationsQuery({
-    variables: { projectId },
-    skip: !githubUserId,
-    ...contextWithCacheHeaders,
-  });
+  const onAcceptInvite = () => {
+    mutate(undefined);
```
Comment (haydencleary) [inline src/components/ProjectLeadInvitation/ProjectLeadInvitation.tsx:22]: Necessary to pass `undefined` ?

File: src/api/useMutationAlert.ts [hunk 5]
```diff
@@ -30,11 +32,12 @@ interface Props {
  * @param {string} props.success.message - The success message to display in the toast.
  */
 export const UseMutationAlert = ({ mutation, error, success }: Props) => {
```
Comment (haydencleary) [inline src/api/useMutationAlert.ts:34]:
```suggestion
export const useMutationAlert = ({ mutation, error, success }: Props) => {
```

File: src/components/ProjectLeadInvitation/ProjectLeadInvitationView.tsx [hunk 6]
```diff
@@ -54,7 +56,12 @@ export default function ProjectLeadInvitationView({
       >
         {projectName ? T("projectLeadInvitation.prompt", { projectName }) : T("project.projectLeadInvitation.prompt")}
       </div>
-      <Button size={ButtonsSizes[size]} onClick={onClick} data-testid="accept-invite-button flex-1 sm:flex-auto">
+      <Button
+        size={ButtonsSizes[size]}
+        onClick={onClick}
+        data-testid="accept-invite-button flex-1 sm:flex-auto"
```
Comment (haydencleary) [inline src/components/ProjectLeadInvitation/ProjectLeadInvitationView.tsx:62]: Normal that tailwind classes are in here ?

Comment (alexbeno) [reply src/components/ProjectLeadInvitation/ProjectLeadInvitationView.tsx:62]: i think it's for testing purpose, i only touch this component to add a disabled props :/

**Review by haydencleary — APPROVED** [review]

**Pull request merged**